qfieldsync/core/offline_converter.py

```diff
@@ -155,6 +155,12 @@ def convert(self):
                             'Both "Area of Interest" and "only selected features" options were enabled, tha latter takes precedence.'),
                             'QFieldSync')
                     self.__offline_layers.append(layer)
+
+                    # Store the primary key field name(s) as comma separated custom property
+                    if layer.type() == QgsMapLayer.VectorLayer:
+                        key_fields = ','.join([layer.fields()[x].name() for x in layer.primaryKeyAttributes()])
+                        layer.setCustomProperty('QFieldSync/cloudPrimaryKeys', key_fields)
+
```

Comment:
Can we rename this to "sourceDataPrimaryKey" or similar? Since it's qfieldsync (and not qfieldcloud only).

Reply:
ok